extensions/itunes/init.lua

```diff
@@ -281,6 +281,20 @@ function itunes.setPosition(p)
   return tell('set player position to '..p)
 end
 
+--- hs.itunes.getDuration()
+--- Function
+--- Gets the duration (in seconds) of the current song
+---
+--- Parameters:
+---  * None
+---
+--- Returns:
+---  * The number of seconds long the current song is, 0 if no song is playing
+function itunes.getDuration()
+  local duration = tonumber(tell('duration of current track'))
+  return duration ~= nil and duration or 0
```

Comment:
I can't think of a situation where a song will ever be 0 seconds long - seems like it that would be a broken file, no?

The reason why I converted the nil to a 0 was so that the method always returned a consistent type. That means that I could operate directly with `hs.itunes.getDuration()` without having to worry about exceptions. Like: `local minutes = hs.itunes.getDuration() / 60`

Happy to change it if you think returning nil is better though!